util/image_viewer/spotfinder_frame.py

```diff
@@ -1693,7 +1693,11 @@ def map_coords(x, y, p):
                             continue
                         x, y = detector[panel].get_ray_intersection_px(s1)
                         x, y = map_coords(x, y, panel)
-                        vector_data.append((((beam_x, beam_y), (x, y)), vector_dict))
+                        basis_vector_dict = dict(vector_dict)
+                        basis_vector_dict["color"] = self.prediction_colours[i_expt]
+                        vector_data.append(
+                            (((beam_x, beam_y), (x, y)), basis_vector_dict)
+                        )
```

Comment:
I think this should work, without the need to explicitly create a modified copy of `vector_dict`:
```suggestion
                        vector_data.append(
                            (((beam_x, beam_y), (x, y)), {**vector_dict, "color": self.prediction_colours[i_expt]}),
                        )
```

Reply:
:+1: I'll commit and try it